cmd/dep/status.go

```diff
@@ -641,7 +660,18 @@ func runStatusAll(ctx *dep.Ctx, out outputter, p *dep.Project, sm gps.SourceMana
 
 		// Use the collected BasicStatus in outputter.
 		for _, proj := range slp {
-			if err := out.BasicLine(bsMap[string(proj.Ident().ProjectRoot)]); err != nil {
+			pr := proj.Ident().ProjectRoot
+
+			// If -old flag, only display the lines where the solver mismatches
+			if cmd.old {
+				if matches, err := projUpToDate(proj, solution); matches {
+					continue
+				} else if err != nil {
+					return false, 0, err
+				}
+			}
+
+			if err := out.BasicLine(bsMap[string(pr)]); err != nil {
```

Comment:
Let's not touch this function at all. This whole function would be refactored into small functions in https://github.com/golang/dep/pull/1498 , and that would make things a lot better :) . So, let's move all these changes into a separate function `runOld`.